latency.go

```diff
@@ -19,17 +19,61 @@ import (
 	"github.com/tylertreat/hdrhistogram-writer"
 )
 
+// Test Parameters
 var (
-	ServerA, ServerB string
-	TargetPubRate    int
-	MsgSize          int
-	NumPubs          int
-	TestDuration     time.Duration
-	HistFile         string
+	ServerA       string
+	ServerB       string
+	TargetPubRate int
+	MsgSize       int
+	NumPubs       int
+	TestDuration  time.Duration
+	HistFile      string
+	Secure        bool
+	TLSca         string
+	TLSkey        string
+	TLScert       string
 )
 
+var usageStr = `
+Usage: latency-tests [options]
+
+Test Options:
+    -sa <url>        ServerA (Publish) (default: nats://localhost:4222)
+    -sb <url>        ServerB (Subscribe) (default: nats://localhost:4222)
+    -sz <int>        Message size in bytes (default: 8)
+    -tr <int>        Rate in msgs/sec (default: 1000)
+    -tt <string>     Test duration (default: 5s)
+    -hist <file>     Histogram file
+    -secure          Enable TLS without verfication (default: false)
+    -tls_ca <string> TLS Certificate CA file
+    -tls_key <file>  TLS Private Key
+    -tls_cert <file> TLS Certificate`
+
 func usage() {
-	log.Fatalf("Usage: latency [-sa serverA] [-sb serverB] [-sz msgSize] [-tr msgs/sec] [-tt testTime] [-hist <file>]\n")
+	log.Fatalf(usageStr + "\n")
+}
+
+// connect creates a connection to a NATS server. TLS options are set as
+// specified by command line pamameters
+func connect(url string) (*nats.Conn, error) {
+	opts := nats.GetDefaultOptions()
```

Comment:
These can all be options to a nc.Connect() call. Prefer not to do the opts.Connect() anymore. Let's use the more modern way here.